src/main/java/github/weichware10/util/db/Configurations.java

```diff
@@ -0,0 +1,141 @@
+            boolean tutorial = (rs.getInt("tutorial") == 1) ? true : false;
+
+            if (toolType == ToolType.CODECHARTS) {
+                // CODECHARTS spezifische Werte
+                List<String> strings = Util.stringsToList(rs.getString("strings"));
+                int[] initialSize = new int[] {
+                        rs.getInt("initialsize_x"),
+                        rs.getInt("initialsize_y") };
+                long[] timings = new long[] { rs.getLong("timings0"), rs.getLong("timings1") };
+
+                // CodeChartsConfiguration erstellen
+                CodeChartsConfiguration codeChartsConfiguration = new CodeChartsConfiguration(
+                        strings, initialSize, timings, tutorial, imageUrls);
+
+                // komplette Konfiguration zurückgeben
+                return new Configuration(configId, codeChartsConfiguration);
+            } else {
+                // ZOOMMAPS spezifische Werte
+                float speed = rs.getFloat("speed");
+
+                // ZoomMapsConfiguration erstellen
+                ZoomMapsConfiguration zoomMapsConfiguration = new ZoomMapsConfiguration(
+                        speed, tutorial, imageUrls);
+
+                // komplette Konfiguration zurückgeben
+                return new Configuration(configId, zoomMapsConfiguration);
+            }
+        } catch (SQLException e) {
+            Logger.error("SQLException when processing query result", e);
+            return null;
+        }
+    }
+
+    public String setConfiguration(Configuration configuration) {
+        final String ccQuery = """
+                INSERT INTO configurations
+                (configid, tooltype, tutorial, question, imageurls,
+                strings, initialsize_x, initialsize_y, timings_0, timings_1, speed)
+                VALUES
+                ('%s', '%s', %d, '%s', '%s',
+                '%s', %d, %d, %d, %d, null)""";
+
+        final String zmQuery = """
+                INSERT INTO configurations
+                (configid, tooltype, tutorial, question, imageurls,
+                strings, initialsize_x, initialsize_y, timings_0, timings_1, speed)
+                VALUES
+                ('%s', '%s', %d, '%s', '%s',
+                null, null, null, null, null, %f)""";
+
+        final String uniqueException =
+                "ERROR: duplicate key value violates unique constraint \"configurations_pkey\"";
+
+        boolean idUnique = false;
+        String id = null;
+
+        while (!idUnique) {
+            idUnique = true;
+            id = Util.generateId("con_", 5);
+            try {
+                Connection conn = DriverManager.getConnection(url, props);
+                Statement st = conn.createStatement();
+                if (configuration.getToolType() == ToolType.CODECHARTS) {
+                    st.executeQuery(String.format(zmQuery, id, "CODECHARTS", 1, "tst", "tst", 3.0f));
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 101).